alws/utils/noarch.py

```diff
@@ -0,0 +1,127 @@
+import copy
+import typing
+
+import sqlalchemy
+from sqlalchemy.future import select
+from sqlalchemy.orm import Session, selectinload
+
+from alws import models
+from alws.config import settings
+from alws.constants import BuildTaskStatus
+from alws.utils.pulp_client import PulpClient
+
+
+__all__ = [
+    'get_noarch_packages',
+    'save_noarch_packages',
+]
+
+
+async def get_noarch_packages(
+    db: Session,
+    build_task_ids: typing.List[int]
+) -> typing.Tuple[dict, dict]:
+    query = select(models.BuildTaskArtifact).where(sqlalchemy.and_(
+        models.BuildTaskArtifact.build_task_id.in_(build_task_ids),
+        models.BuildTaskArtifact.type == 'rpm',
+        models.BuildTaskArtifact.name.like('%.noarch.%'),
+    ))
+    db_artifacts = await db.execute(query)
+    db_artifacts = db_artifacts.scalars().all()
+    noarch_packages = {}
+    debug_noarch_packages = {}
+    for artifact in db_artifacts:
+        if '-debuginfo-' in artifact.name or '-debugsource-' in artifact.name:
```

Comment:
Maybe you will like the next construction: `any((debug in artifact.name for debug in ('-debuginfo-', '-debugsource-')))`